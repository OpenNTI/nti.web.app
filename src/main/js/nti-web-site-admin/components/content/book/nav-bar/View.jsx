import React from 'react';

import Card from '../../../common/Card';

import Identity from './Identity';
import Tabs from './Tabs';

export default function SiteAdminBookNavBar (props) {
	return (
		<Card className="site-admin-course-nav-bar">
			<Identity {...props} />
			<Tabs {...props} />
		</Card>
	);
}
