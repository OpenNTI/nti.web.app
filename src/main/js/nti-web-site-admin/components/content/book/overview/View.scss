@import '~nti-style-common';

.site-admin-book-overview {
	@include clearfix;

	.site-admin-active-days {
		width: 100%;
		margin-top: 5px;
	}

	.active-users {
		width: 100%;

		.dashboard-list-widget {
			margin: 0 0 10px;
		}
	}

	.active-times-widget {
		margin: 0 0 10px;
	}

	.site-admin-active-days {
		margin: 10px 0;
	}

	.site-admin-row {
		display: flex;

		.site-admin-card {
			width: 250px;
			height: 320px;

			.value {
				padding-top: 1rem;
			}
		}

		.active-users {
			height: 320px;

			.header {
				margin-top: 0;
				margin-bottom: 0;
			}
		}
	}
}
