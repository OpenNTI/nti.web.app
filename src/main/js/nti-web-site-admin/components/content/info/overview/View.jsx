import PropTypes from 'prop-types';

import { AdminTools } from '@nti/web-course';

SiteAdminCourseOverview.propTypes = {
	course: PropTypes.object,
};
export default function SiteAdminCourseOverview({ course }) {
	return <AdminTools.Dashboard course={course} />;
}
