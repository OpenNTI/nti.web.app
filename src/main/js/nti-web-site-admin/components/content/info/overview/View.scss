@import '~nti-style-common';

.site-admin-course-overview {
	@include clearfix;

	.site-admin-active-days {
		width: 100%;
		margin-top: 5px;
	}

	.site-admin-row {
		display: flex;

		.site-admin-card {
			width: 250px;
			height: 320px;

			.value {
				padding-top: 1rem;
			}
		}

		.active-users {
			margin: 0 0 0 4px;
			height: 320px;

			.header {
				margin-top: 0;
				margin-bottom: 0;
			}
		}
	}
}
