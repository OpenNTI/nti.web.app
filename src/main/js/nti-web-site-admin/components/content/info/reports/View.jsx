import React from 'react';
import PropTypes from 'prop-types';

import { List } from '@nti/web-reports';

import Card from '../../../common/Card';

SiteAdminCourseReports.propTypes = {
	course: PropTypes.object,
};
export default function SiteAdminCourseReports({ course }) {
	return course ? (
		<Card>
			<List context={course} />
		</Card>
	) : null;
}
