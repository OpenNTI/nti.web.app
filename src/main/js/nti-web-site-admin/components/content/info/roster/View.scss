@import '~@nti/style-common';

.nti-site-admin-course-roster {
	.site-admin-course-roster-header {
		display: flex;
		flex-direction: row;
		justify-content: flex-end;
		margin-bottom: 1rem;
	}
}

.site-admin-course-roster {
	.loading-mask {
		position: relative;
		min-height: 200px;
	}

	ul {
		li {
			box-shadow: 0 1px 0 0 var(--border-grey)-light;

			a {
				text-decoration: none;
				display: block;
			}
		}
	}

	.controls {
		span {
			color: var(--primary-blue);
			margin-right: 10px;
			cursor: pointer;
		}
	}
}
