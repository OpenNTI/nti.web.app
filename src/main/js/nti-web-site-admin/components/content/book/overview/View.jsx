import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { Widgets } from '@nti/web-reports';

const { ActiveDays, ActiveUsers, ActiveTimes } = Widgets;

export default class SiteAdminBookOverview extends React.Component {
	static propTypes = {
		course: PropTypes.object
	}

	state = {}

	componentDidMount () {
		const {course} = this.props;

		if(!this.state.resp && course && course.hasLink('users')) {
			course.fetchLink('users').then(resp => {
				this.setState({resp});
			});
		}
	}

	render () {
		const {course} = this.props;

		return (
			<div className="site-admin-book-overview">
				<div className="site-admin-row">
					<div className="active-users">
						<ActiveUsers entity={course}/>
					</div>
				</div>
				<div className="active-days">
					<ActiveDays entity={course}/>
				</div>
				<div className="active-times">
					<ActiveTimes course={course}/>
				</div>
			</div>
		);
	}
}
