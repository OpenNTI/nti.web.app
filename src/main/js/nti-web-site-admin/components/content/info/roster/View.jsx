import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {Roster, Enrollment} from '@nti/web-course';
import {List, Loading, Button} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';
import {LinkTo} from '@nti/web-routing';

import ErrorMessage from '../../../common/ErrorMessage';
import Card from '../../../common/Card';

import Item from './Item';

const DEFAULT_TEXT = {
	error: 'Unable to load roster.',
	manage: 'Manage Roster'
};
const t = scoped('nti-site-admin.courses.course.info.Roster', DEFAULT_TEXT);

export default class SiteAdminCourseRoster extends React.Component {
	static propTypes = {
		course: PropTypes.object
	}


	attachRoster = x => this.roster = x;

	onChange = () => {
		if (this.roster) {
			this.roster.reload();
		}
	}


	render () {
		const {course} = this.props;

		return (
			<div className="nti-site-admin-course-roster">
				{this.renderHeader(course)}
				<Roster ref={this.attachRoster} course={course} renderRoster={this.renderRoster} />
			</div>
		);
	}


	renderRoster = ({loading, error, items, loadNextPage, loadPrevPage}) => {
		return (
			<div className="site-admin-course-roster">
				{loading && (<div className="loading-mask"><Loading.Mask /></div>)}
				{error && (<ErrorMessage>{t('error')}</ErrorMessage>)}
				{!loading && this.renderItems(items)}
				{!loading && this.renderControls(loadPrevPage, loadNextPage)}
			</div>
		);
	}


	renderHeader (course) {
		const {CatalogEntry} = course;

		return (
			<div className="site-admin-course-roster-header">
				<Enrollment.Admin.Prompt.Trigger course={CatalogEntry} onChange={this.onChange}>
					<Button rounded>
						{t('manage')}
					</Button>
				</Enrollment.Admin.Prompt.Trigger>
			</div>
		);
	}


	renderItems (items) {
		return (
			<Card>
				<List.Unadorned>
					{items.map((item, index) => {
						return (
							<li key={index}>
								<LinkTo.Object object={item} context="site-admin.courses.course-roster.list">
									<Item item={item} onChange={this.onChange} />
								</LinkTo.Object>
							</li>
						);
					})}
				</List.Unadorned>
			</Card>
		);
	}


	renderControls (loadPrevPage, loadNextPage) {
		return (
			<div className="controls">
				{loadPrevPage && (<span onClick={loadPrevPage}>Prev</span>)}
				{loadNextPage && (<span onClick={loadNextPage}>Next</span>)}
			</div>
		);
	}
}
