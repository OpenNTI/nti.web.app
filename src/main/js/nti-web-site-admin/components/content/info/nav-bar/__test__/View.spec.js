import React from 'react';
import renderer from 'react-test-renderer';

import View from '../View';

const getCourse = (hasRosterLink) => {
	return {
		CatalogEntry: {
			Title: 'course1',
			getDefaultAssetRoot () { return 'testRoot'; }
		},
		getPresentationProperties: () => {
			return {
				title: 'course1',
				label: 'Course 1'
			};
		},
		hasLink: () => hasRosterLink
	};
};

/* eslint-env jest */
describe('Site admin user course enrollment nav bar test', () => {
	test('Basic render test (has roster link)', async () => {
		const course = getCourse(true);

		const cmp = renderer.create(<View course={course}/>);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Basic render test (has no roster link)', async () => {
		const course = getCourse(false);

		const cmp = renderer.create(<View course={course}/>);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
