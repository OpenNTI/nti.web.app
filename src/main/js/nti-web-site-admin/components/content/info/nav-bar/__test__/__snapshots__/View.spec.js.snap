// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user course enrollment nav bar test Basic render test (has no roster link) 1`] = `
<div
  className="site-admin-card site-admin-course-nav-bar"
>
  <div
    className="site-admin-course-identity"
  >
    <img
      className="course-icon"
      src=""
    />
    <div
      className="info"
    >
      <div
        className="label"
      >
        Course 1
      </div>
      <div
        className="title"
      >
        course1
      </div>
    </div>
    <div
      className="actions"
    >
      <div
        className="nti-link-to-path profile"
        to="#"
      >
        <i
          className="icon-goto"
        />
        <span>
          View Course
        </span>
      </div>
    </div>
  </div>
  <ul
    className="site-admin-tabs"
  >
    <li>
      <div
        className="nti-link-to-path"
        exact={true}
        to="./"
      >
        Overview
      </div>
    </li>
    <li />
    <li>
      <div
        className="nti-link-to-path"
        to="./reports"
      >
        Reports
      </div>
    </li>
  </ul>
</div>
`;

exports[`Site admin user course enrollment nav bar test Basic render test (has roster link) 1`] = `
<div
  className="site-admin-card site-admin-course-nav-bar"
>
  <div
    className="site-admin-course-identity"
  >
    <img
      className="course-icon"
      src=""
    />
    <div
      className="info"
    >
      <div
        className="label"
      >
        Course 1
      </div>
      <div
        className="title"
      >
        course1
      </div>
    </div>
    <div
      className="actions"
    >
      <div
        className="nti-link-to-path profile"
        to="#"
      >
        <i
          className="icon-goto"
        />
        <span>
          View Course
        </span>
      </div>
    </div>
  </div>
  <ul
    className="site-admin-tabs"
  >
    <li>
      <div
        className="nti-link-to-path"
        exact={true}
        to="./"
      >
        Overview
      </div>
    </li>
    <li>
      <div
        className="nti-link-to-path"
        to="./roster"
      >
        Roster
      </div>
    </li>
    <li>
      <div
        className="nti-link-to-path"
        to="./reports"
      >
        Reports
      </div>
    </li>
  </ul>
</div>
`;
